apps/aot/src/app/events/(leaderboard)/_components/daily-leaderboard.tsx

```diff
@@ -4,7 +4,7 @@ import { getAotChallengeIdForAdventDay } from './getAotChallengeIds';
 
 export const getFirst100SubmissionsRanked = async (adventYear: string, adventDay: string) => {
   // We already checked adventDay is valid in day/[day]/layout.tsx
-  const challengeId = await getAotChallengeIdForAdventDay(adventYear, adventDay)!;
+  const challengeId = await getAotChallengeIdForAdventDay(adventYear, adventDay);
```

Comment:
[`@typescript-eslint/no-unnecessary-type-assertion`](https://typescript-eslint.io/rules/no-unnecessary-type-assertion): the `!` here was applying to the `getAotChallengeIdForAdventDay(...)`, not the `await ...`. `challengeId` is `number | undefined` either way.